Lib/email/utils.py

```diff
@@ -106,12 +106,52 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _pre_parse_validation(email_header_fields):
+    accepted_values = []
+    for v in email_header_fields:
+        s = v.replace('\\(', '').replace('\\)', '')
+        if s.count('(') != s.count(')'):
+            v = "('', '')"
+        accepted_values.append(v)
+
+    return accepted_values
+
+
+def _post_parse_validation(parsed_email_header_tuples):
+    accepted_values = []
+    # The parser would have parsed a correctly formatted domain-literal
+    # The existence of an [ after parsing indicates a parsing failure
+    for v in parsed_email_header_tuples:
+        if '[' in v[1]:
+            v = ('', '')
+        accepted_values.append(v)
+
+    return accepted_values
+
 
 def getaddresses(fieldvalues):
-    """Return a list of (REALNAME, EMAIL) for each fieldvalue."""
-    all = COMMASPACE.join(str(v) for v in fieldvalues)
+    """Return a list of (REALNAME, EMAIL) or ('','') for each fieldvalue.
+    
+    When parsing fails for a fieldvalue, a 2-tuple of ('', '') is returned in
+    its place.
+
+    If the resulting list is greater than number of items in the fieldvalues
```

Comment:
From the standpoint of someone calling this function, this sentence doesn't make much sense. They'll wonder how this could ever happen and it isn't clear what "the resulting list" means in this case such that it could.  this is an implementation detail as described.

I think what you're trying to say is that if an internal disagreement in how to parse the email addresses into individual results arises, the `[('', '')]` value is returned instead of potentially invalid information. It'd be good to describe when this scenario might occur to the user in a plausible manner.

Reply:
Will do. Got busy with work. Hopefully, I'll have this change made by the end of the week and figure out what the issue is with that 1 unit test that is failing. Odd thing is that all of the Unit tests pass when I run them locally on Ubuntu.